src/Cli/dotnet/PerformanceLogManager.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.ShellShim;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal sealed class PerformanceLogManager
```

Comment:
again, try to reuse https://github.com/dotnet/sdk/blob/b1223209644d900702287faea8e9b71f95ec49f8/src/Cli/dotnet/Telemetry/PersistenceChannel/StorageService.cs